### ISSUE TITLE
Can't access LAN or WAN when connected to VPN

### ISSUE THREAD

**Issue by nitvit610** [pivpn/pivpn#753]
## In raising this issue, I confirm the following: 

`{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [] The issue I am reporting can be *replicated*.
- [] The issue I am reporting can be *is* directly related to the pivpn installer script.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

### Issue
Set up PiVPN and PiHole in Google Cloud following rajannpatel guide:
https://github.com/rajannpatel/Pi-Hole-PiVPN-on-Google-Compute-Engine-Free-Tier-with-Full-Tunnel-and-Split-Tunnel-OpenVPN-Configs

All clients connect fine and DNS works etc, but my Android phone does not. 
The phone previously worked fine until this morning (no changes).
Pages load when not connected to VPN but as soon as I connect it stops working.
I don't see any issues in OVPN logs and I can see queries in PiHole.

### Have you searched for similar issues and solutions?
    (yes)
Looked through loads of other pages online but have not found one that has helped so far.

### Console output of      `curl -L install.pivpn.io | bash`
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   178  100   178    0     0    851      0 --:--:-- --:--:-- --:--:--   855
100 54916  100 54916    0     0   128k      0 --:--:-- --:--:-- --:--:--  128k
:::
::: sudo will be used for the install.
::: Update option selected.
::: Verifying free disk space...
:::
::: Checking apt-get for upgraded packages.... done!
:::
::: There are 4 updates available for your system!
::: We recommend you update your OS after installing PiVPN! 
:::
::: Using IP address: 10.142.0.3
::: Using interface: eth0
::: Using User: nitvit610
::: Using protocol: udp
::: Using port: 1194
:::
::: IP Information
::: Since we think you are not using Raspbian, we will not configure a static IP for you.
::: If you are in Amazon then you can not configure a static IP anyway.
::: Just ensure before this installer started you had set an elastic IP on your instance.
:::
::: Checking for existing base files...
:::    Checking /etc/.pivpn is a repo... OK!
:::     Updating repo in /etc/.pivpn...fatal: repository '' does not exist
main: line 565: cd: /etc/.pivpn: No such file or directory
```

### Console output of      `pivpn add` or `pivpn add nopass`
```
Enter a Name for the Client:  test
How many days should the certificate last?  1080
spawn ./easyrsa build-client-full test nopass

Note: using Easy-RSA configuration from: ./vars

Using SSL: openssl OpenSSL 1.1.0j  20 Nov 2018
Generating an EC private key
writing new private key to '/etc/openvpn/easy-rsa/pki/private/test.key.EoQHtEogCv'
-----
Using configuration from /etc/openvpn/easy-rsa/pki/safessl-easyrsa.cnf
Check that the request matches the signature
Signature ok
The Subject's Distinguished Name is as follows
commonName            :ASN.1 12:'test'
Certificate is to be certified until May  6 13:09:10 2022 GMT (1080 days)

Write out database with 1 new entries
Data Base Updated
Client's cert found: test.crt
Client's Private Key found: test.key
CA public Key found: ca.crt
tls-auth Private Key found: ta.key


========================================================
Done! test.ovpn successfully created! 
test.ovpn was copied to:
  /home/nitvit610/ovpns
for easy transfer. Please use this profile only on one
device and create additional profiles for other devices.
========================================================
```

### Console output of      `pivpn debug`
```
::: Generating Debug Output

::::            PiVPN debug              ::::
=============================================
::::            Latest commit            ::::
fatal: Not a git repository: '/etc/.pivpn/.git'
=============================================
::::        Installation settings        ::::
/etc/pivpn/DET_PLATFORM -> Debian
/etc/pivpn/INSTALL_PORT -> 1194
/etc/pivpn/INSTALL_PROTO -> udp
/etc/pivpn/INSTALL_USER -> nitvit610
/etc/pivpn/NO_UFW -> 1
/etc/pivpn/pivpnINTERFACE -> eth0
/etc/pivpn/TWO_POINT_FOUR -> 
=============================================
::::    setupVars file shown below       ::::
pivpnUser=nitvit610
UNATTUPG=unattended-upgrades
pivpnInterface=eth0
IPv4dns=
IPv4addr=10.142.0.3
IPv4gw=10.142.0.1
pivpnProto=udp
PORT=1194
ENCRYPT=256
APPLY_TWO_POINT_FOUR=true
DOWNLOAD_DH_PARAM=false
PUBLICDNS=
OVPNDNS1=10.8.0.1
OVPNDNS2=
=============================================
::::  Server configuration shown below   ::::
dev tun
proto udp
port 1194
ca /etc/openvpn/easy-rsa/pki/ca.crt
cert /etc/openvpn/easy-rsa/pki/issued/server_aEWPKAojgRoczDoR.crt
key /etc/openvpn/easy-rsa/pki/private/server_aEWPKAojgRoczDoR.key
dh none
topology subnet
server 10.8.0.0 255.255.255.0
# Set your primary domain name server address for clients
push "dhcp-option DNS 10.8.0.1"
push "block-outside-dns"
# Override the Client default gateway by using 0.0.0.0/1 and
# 128.0.0.0/1 rather than 0.0.0.0/0. This has the benefit of
# overriding but not wiping out the original default gateway.
# push "redirect-gateway def1"
client-to-client
# keepalive 1800 3600
keepalive 10 60
remote-cert-tls client
tls-version-min 1.2
tls-crypt /etc/openvpn/easy-rsa/pki/ta.key
# cipher AES-256-CBC
cipher AES-128-GCM
auth SHA256
user nobody
group nogroup
persist-key
persist-tun
crl-verify /etc/openvpn/crl.pem
status /var/log/openvpn-status.log 20
status-version 3
syslog
verb 3
#DuplicateCNs allow access control on a less-granular, per user basis.
#Remove # if you will manage access by user instead of device. 
#duplicate-cn
# Generated for use by PiVPN.io
# performance stuff
fast-io
compress lz4-v2
push "compress lz4-v2"
=============================================
::::  Client template file shown below   ::::
client
dev tun
proto udp
remote REMOTE 1194
resolv-retry infinite
nobind
persist-key
persist-tun
remote-cert-tls server
tls-version-min 1.2
verify-x509-name server_aEWPKAojgRoczDoR name
cipher AES-256-CBC
auth SHA256
auth-nocache
verb 3
=============================================
::::    Recursive list of files in       ::::
::: /etc/openvpn/easy-rsa/pki shows below :::
/etc/openvpn/easy-rsa/pki/:
ca.crt
crl.pem
Default.txt
ecparams
extensions.temp
index.txt
index.txt.attr
index.txt.attr.old
index.txt.old
issued
openssl-easyrsa.cnf
private
renewed
revoked
safessl-easyrsa.cnf
serial
serial.old
ta.key
test.ovpn

/etc/openvpn/easy-rsa/pki/ecparams:
prime256v1.pem
/etc/openvpn/easy-rsa/pki/issued:
server_aEWPKAojgRoczDoR.crt
test.crt

/etc/openvpn/easy-rsa/pki/private:
ca.key
server_aEWPKAojgRoczDoR.key
test.key

/etc/openvpn/easy-rsa/pki/renewed:
private_by_serial
reqs_by_serial

/etc/openvpn/easy-rsa/pki/renewed/private_by_serial:

/etc/openvpn/easy-rsa/pki/renewed/reqs_by_serial:

/etc/openvpn/easy-rsa/pki/revoked:
private_by_serial
reqs_by_serial

/etc/openvpn/easy-rsa/pki/revoked/private_by_serial:
65301FE6ACDE7F94DFA8DACF5ED4C783.key
93FF682BE52E29EBF676A75D21B5E8D1.key

/etc/openvpn/easy-rsa/pki/revoked/reqs_by_serial:
65301FE6ACDE7F94DFA8DACF5ED4C783.req
93FF682BE52E29EBF676A75D21B5E8D1.req
=============================================
::::            Self check               ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] OpenVPN is running
:: [OK] OpenVPN is enabled (it will automatically start on reboot)
:: [OK] OpenVPN is listening on port 1194/udp
=============================================
::::      Snippet of the server log      ::::
May 22 14:06:47 pi-hole ovpn-server[600]: 90.213.120.103:60267 peer info: IV_NCP=2
May 22 14:06:47 pi-hole ovpn-server[600]: 90.213.120.103:60267 peer info: IV_TCPNL=1
May 22 14:06:47 pi-hole ovpn-server[600]: 90.213.120.103:60267 peer info: IV_PROTO=2
May 22 14:06:47 pi-hole ovpn-server[600]: 90.213.120.103:60267 peer info: IV_IPv6=0
May 22 14:06:47 pi-hole ovpn-server[600]: 90.213.120.103:60267 peer info: IV_AUTO_SESS=1
May 22 14:06:47 pi-hole ovpn-server[600]: 90.213.120.103:60267 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1550', remote='link-mtu 1521'
May 22 14:06:47 pi-hole ovpn-server[600]: 90.213.120.103:60267 WARNING: 'comp-lzo' is present in local config but missing in remote config, local='comp-lzo'
May 22 14:06:47 pi-hole ovpn-server[600]: 90.213.120.103:60267 WARNING: 'auth' is used inconsistently, local='auth [null-digest]', remote='auth SHA256'
May 22 14:06:47 pi-hole ovpn-server[600]: 90.213.120.103:60267 Control Channel: TLSv1.2, cipher TLSv1/SSLv3 ECDHE-ECDSA-AES256-GCM-SHA384
May 22 14:06:47 pi-hole ovpn-server[600]: 90.213.120.103:60267 [useriphone] Peer Connection Initiated with [AF_INET]90.213.120.103:60267
May 22 14:06:47 pi-hole ovpn-server[600]: useriphone/90.213.120.103:60267 MULTI_sva: pool returned IPv4=10.8.0.4, IPv6=(Not enabled)
May 22 14:06:47 pi-hole ovpn-server[600]: useriphone/90.213.120.103:60267 MULTI: Learn: 10.8.0.4 -> useriphone/90.213.120.103:60267
May 22 14:06:47 pi-hole ovpn-server[600]: useriphone/90.213.120.103:60267 MULTI: primary virtual IP for useriphone/90.213.120.103:60267: 10.8.0.4
May 22 14:06:47 pi-hole ovpn-server[600]: useriphone/90.213.120.103:60267 PUSH: Received control message: 'PUSH_REQUEST'
May 22 14:06:47 pi-hole ovpn-server[600]: useriphone/90.213.120.103:60267 SENT CONTROL [useriphone]: 'PUSH_REPLY,dhcp-option DNS 10.8.0.1,block-outside-dns,compress lz4-v2,route-gateway 10.8.0.1,topology subnet,ping 10,ping-restart 60,ifconfig 10.8.0.4 255.255.255.0,peer-id 2,cipher AES-256-GCM' (status=1)
May 22 14:06:47 pi-hole ovpn-server[600]: useriphone/90.213.120.103:60267 Data Channel Encrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
May 22 14:06:47 pi-hole ovpn-server[600]: useriphone/90.213.120.103:60267 Data Channel Decrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
May 22 14:07:27 pi-hole ovpn-server[600]: 90.213.120.103:53341 TLS Error: TLS key negotiation failed to occur within 60 seconds (check your network connectivity)
May 22 14:07:27 pi-hole ovpn-server[600]: 90.213.120.103:53341 TLS Error: TLS handshake failed
May 22 14:07:27 pi-hole ovpn-server[600]: 90.213.120.103:53341 SIGUSR1[soft,tls-error] received, client-instance restarting
=============================================
::::            Debug complete           ::::
::: 
::: Debug output completed above.
::: Copy saved to /tmp/debug.txt
::: 
```
### Have you taken any steps towards solving your issue?
```
  Have looked at most articles I could find but nothing seemed to really be linked.
Not sure why just my mobile client is affected with all others working.
```

**Comment by orazioedoardo** [comment]
You seem to have customized the server configuration. The default one does not use compression, and as far as i know, since compression introduces a vulnerability, on mobile you have to explicitly enable it in the app settings if you want to use it.

**Comment by nitvit610** [comment]
> You seem to have customized the server configuration. The default one does not use compression, and as far as i know, since compression introduces a vulnerability, on mobile you have to explicitly enable it in the app settings if you want to use it.

Hmm, I can take a look at removing it and see if that changes anything.

I've disabled the compression in the OpenVPN client on all devices, and it is working on another Android, iPhone and Windows 10 fine so it is strange just my Android one is being affected.

I have also tried reinstalling the client on my phone, and revoking/readding the user, with no luck.

Is there anything else I can try?
